src/js/components/ContainerSettingsComponent.jsx

```diff
@@ -0,0 +1,265 @@
+        volumes: [true]
+      }
+    };
+  },
+
+  handleAddRow: function (rowKey, position, event) {
+    event.target.blur();
+    event.preventDefault();
+    var rows = React.addons.update(
+      this.state.rows, {[rowKey]: {$push: [true]}}
+    );
+    this.setState({rows: rows});
+  },
+
+  handleRemoveRow: function (rowKey, position, event) {
+    event.target.blur();
+    event.preventDefault();
+    var rows = React.addons.update(
+      this.state.rows, {[rowKey]: {[position]: {$set: false}}}
+    );
+    if (rows[rowKey].filter((exists) => exists).length === 0) {
+      rows[rowKey].push(true);
+    }
+    this.setState({rows: rows});
+  },
+
+  getPortMappingRow: function (i = 0) {
+    return (
+      <div key={`pm-${i}`} className="row duplicable-row">
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].containerPort`}
+            label="Container Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].hostPort`}
+            label="Host Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].servicePort`}
+            label="Service Port"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <input type="number" step="1" min="0" max="65535"/>
+          </FormGroupComponent>
+        </div>
+        <div className="col-sm-3">
+          <FormGroupComponent
+            attribute={`container.docker.portMappings[${i}].protocol`}
+            label="Protocol"
+            model={this.props.model}
+            errors={this.props.errors}
+            validator={appValidator}>
+            <select>
+              <option value="" disabled="disabled" selected="selected">Select</option>
+              <option value={ContainerConstants.PORTMAPPINGS.PROTOCOL.TCP}>
+                          {ContainerConstants.PORTMAPPINGS.PROTOCOL.TCP}
+              </option>
+              <option value={ContainerConstants.PORTMAPPINGS.PROTOCOL.UDP}>
+                          {ContainerConstants.PORTMAPPINGS.PROTOCOL.UDP}
+              </option>
+            </select>
+          </FormGroupComponent>
+          <DuplicableRowControls
+            handleAddRow={this.handleAddRow.bind(null, "portMappings", i)}
+            handleRemoveRow={this.handleRemoveRow.bind(null, "portMappings", i)} />
```

Comment:
The binding on a callback passed as a property as it's done here is the actual documented way of doing things in the React docs.  Take a look at [this](https://facebook.github.io/react/tips/communicate-between-components.html):

> Notice the use of bind(this, arg1, arg2, ...): we're simply passing more arguments to handleClick. This is not a new React concept; it's just JavaScript.

I reckon passing `null` instead of `this` may seem a bit weird, so we could change that (the effect is unchanged). 

In general I tend to favor the `.bind()` over the proposed suggestions as they don't seem to add much value; to the contrary, they would introduce new patterns (e.g. passing function arguments as properties, using the DOM to resolve a simple callback, etc) with the ultimate result of complicating things unnecessarily in my opinion. 

If you disagree, that's fine: let us move that discussion outside of this PR, though.